src/library_wasmfs.js

```diff
@@ -49,23 +49,24 @@ FS.createPreloadedFile = FS_createPreloadedFile;
         throw new Error('Invalid encoding type "' + opts.encoding + '"');
       }
 
-      var pathName = stringToNewUTF8(path);
+      return withStackSave(() => {
+        var pathName = stringToUTF8OnStack(path);
 
-      // Copy the file into a JS buffer on the heap.
-      var buf = __wasmfs_read_file(pathName);
-      // The signed integer length resides in the first 8 bytes of the buffer.
-      var length = {{{ makeGetValue('buf', '0', 'i53') }}};
+        // Copy the file into a JS buffer on the heap.
+        var buf = __wasmfs_read_file(pathName);
```

Comment:
Can you close the `withStackSave` block here and do `var buf = withStackSave(...`.  i.e. keep the scope of the `withStackSave` to a minimum?